services/wfm/src/main/java/org/openkilda/wfm/topology/flowhs/utils/RerouteRetryQueue.java

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.utils;
+
+import org.openkilda.model.PathId;
+import org.openkilda.wfm.topology.flowhs.model.FlowRerouteFact;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+public class RerouteRetryQueue {
+    private FlowRerouteFact active = null;
+    private FlowRerouteFact pending = null;
+
+    /**
+     * Add request into "queue".
+     */
+    public void add(FlowRerouteFact reroute) {
+        if (active == null) {
+            active = reroute;
+        } else if (pending == null) {
+            pending = reroute;
+        } else {
+            pending = mergePending(pending, reroute);
+        }
+    }
+
+    /**
+     * Return first/active queue entry.
+     */
+    public Optional<FlowRerouteFact> get() {
+        FlowRerouteFact result = active;
+        if (result == null) {
+            result = pending;
+        }
```

Comment:
As far as I can see state with `active == null and pending != null` is inconsistent for this structure. Maybe we should return only active Fact here for consistency with remove method?

Reply:
or throw exception if `active == null and pending != null`